core/src/main/java/hudson/model/Computer.java

```diff
@@ -164,6 +164,13 @@
     private volatile String cachedHostName;
     private volatile boolean hostNameCached;
 
+    /**
+     * @see #getEnvironment()
+     */
+    private volatile EnvVars cachedEnvironment;
+    private volatile boolean environmentCached;
```

Comment:
@stephenc so you mean `cachedHostName` is wrong as well. Any suggestion for a better pattern ?